From 683a30080dd5b96ba00852ff6a573a6bfd9b61f9 Mon Sep 17 00:00:00 2001
From: Brian Ryner <bryner@google.com>
Date: Tue, 28 Feb 2023 09:36:28 +1100
Subject: [PATCH] Add define to disallow getentropy

---
 crypto/rand/rand_unix.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/crypto/rand/rand_unix.c b/crypto/rand/rand_unix.c
index 0f4525106a..e83610b12d 100644
--- a/crypto/rand/rand_unix.c
+++ b/crypto/rand/rand_unix.c
@@ -370,6 +370,7 @@ static ssize_t syscall_random(void *buf, size_t buflen)
      * internally. So we need to check errno for ENOSYS
      */
 #  if defined(__GNUC__) && __GNUC__>=2 && defined(__ELF__) && !defined(__hpux)
+#    if !defined(NO_GETENTROPY)
     extern int getentropy(void *buffer, size_t length) __attribute__((weak));
 
     if (getentropy != NULL) {
@@ -378,6 +379,7 @@ static ssize_t syscall_random(void *buf, size_t buflen)
         if (errno != ENOSYS)
             return -1;
     }
+#    endif
 #  elif defined(OPENSSL_APPLE_CRYPTO_RANDOM)
     if (CCRandomGenerateBytes(buf, buflen) == kCCSuccess)
 	    return (ssize_t)buflen;
-- 
2.39.2.722.g9855ee24e9-goog

