--- libffi-3.2.1.orig/m4/ax_cc_maxopt.m4	2014-11-12 23:00:59.000000000 +1100
+++ libffi-3.2.1/m4/ax_cc_maxopt.m4	2024-04-11 15:53:12.274904209 +1000
@@ -138,7 +138,7 @@
 
     gnu)
      # default optimization flags for gcc on all systems
-     CFLAGS="-O3 -fomit-frame-pointer"
+     CFLAGS="-fPIC -O3 -fomit-frame-pointer"
 
      # -malign-double for x86 systems
      # LIBFFI -- DON'T DO THIS - CHANGES ABI
